addons/sale_pdf_quote_builder/models/ir_actions_report.py

```diff
@@ -30,12 +30,15 @@ def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
                 has_footer = bool(footer_record.sale_footer)
                 included_product_docs = self.env['product.document']
                 doc_line_id_mapping = {}
+                customer_lang = order.partner_id.lang
+                customer_company_lang = order.partner_id.parent_id.lang
                 for line in order.order_line:
-                    product_product_docs = line.product_id.product_document_ids
-                    product_template_docs = line.product_template_id.product_document_ids
-                    doc_to_include = (
-                        product_product_docs.filtered(lambda d: d.attached_on_sale == 'inside')
-                        or product_template_docs.filtered(lambda d: d.attached_on_sale == 'inside')
+                    product_docs = line.product_id.product_document_ids | line.product_template_id.product_document_ids
+                    doc_to_include = product_docs.filtered(
+                        lambda d: d.attached_on_sale == 'inside' and (
+                            (not customer_lang and d.doc_lang == customer_company_lang) or
+                            (d.doc_lang in (False, customer_lang))
+                        )
```

Comment:
This seems little bit have accessing two o2m field nested for loop also doing filter
I guess you can fetch all documents product wise before order_line or orders for loop and access them here this way it'll improve perf little bit

Reply:
Yes, it would be nice to have it. I will do it right away.